protos/sources/vega/data_source.proto

```diff
@@ -54,6 +56,39 @@ message DataSourceSpecConfiguration {
   repeated vega.data.v1.Filter filters = 2;
 }
 
+// Specifies a data source that derives its content from calling a read method on an Ethereum contract.
+message DataSourceSpecEthCall {
```

Comment:
I think we should be having something like this, no?
```
// EthCallSpecConfiguration type is to be used when handling ETH Oracle data source specs.
// The content of this type is derived from calling an Ethereum read contract.
message EthCallSpecConfiguration {
  // A deterministic id, created from hashing time+address
  string id = 1;
  uint64 time = 2;

  // Ethereum address of the contract
  string address = 3;
  // ABI of the Ethereum contract
  google.protobuf.ListValue abi = 4;
  // Name of the method to call on the contract.
  string method = 5;
  // Arguments used to pass to the method call.
  repeated google.protobuf.Value args = 6;

  // The signers of the data
  repeated vega.data.v1.Signer signers = 7;
  // Simple list of filters used as example, ETH most probably will not use these ones. // ?
  repeated vega.data.v1.EthereumSourceSpecFilter filters = 8;
}
```

Reply:
So one at a time:

```string id = 1;```

We already have an ID on the parent `DataSourceSpec`; or is there something else I'm missing here? e.g. we don't have one for the traditional oracles defined with `DataSourceSpecConfiguration`

```uint64 time = 2;```
What is this time? We've got the trigger mechanism for determining when to make the ethereum calls.

```repeated vega.data.v1.Signer signers = 7;```
Ethereum data sources don't have signers as I understand it. Is there some good reason why we'd want to include some sort of 'no signers' list of signers?

```// Simple list of filters used as example, ETH most probably will not use these ones. // ? repeated vega.data.v1.EthereumSourceSpecFilter filters = 8;
```
We will almost certainly want something like this but until I've got some code that actually makes use of it I figured it made sense to leave it out as it might take some as yet unexpected form (e.g. little scripts that run in an interpreter or what have you)

As for the name I am happy to change it to be consistent - but I can't quite see in my head what the difference is between Spec/Definition/Configuration that seem to be used sort of interchangeably at various levels. Is there some distinction to be made?

Reply:
Yes, I thought we will be saving the time and the ID inside this object, but i might got it wrong.
The Ethereum data should have signers in all cases because it is external

Reply:
The Definition of the data goes from the highest level as `DataSourceDefinition` that can be anything basically in the sub-levels.
`DataSourceSpecConfiguration` is the type left from the previous implementation (which I left for use from the pre-previous) and it is about the current oracles the VEGA system uses.
I suggested the one `EthCallSpecConfiguration` just to be consistent - because these two are going to be on the same level in the tree - just two types of oracles.
`DataSourceSpec` is again left name from the previous implementation and it represents the type of the data that is held by the instrument. If you go and see the Future type - it is `DataSourceSpecForSettlementData` and `DataSourceSpecForTermination` - these are DataSourceDefinition types that are wrapped further as ExternalDataSourceSpec and OracleDataSourceSpec.

Reply:
OK i've renamed it DataSourceSpecConfigurationEthCall to match the others, though I think it's a bit of a mouthful!

I don't understand this:
> The Ethereum data should have signers in all cases because it is external

There aren't any signers for the data submitted by Ethereum oracles. Are you saying there's a good reason to put this in there even though it will always be empty?

Reply:
Yes it is mouthful absolutely. Maybe a better name for both can be found?
Yes I am referring to this conversation- https://vegaprotocol.slack.com/archives/C056V8CBSCT/p1684239811515819
It just does not have to be signed with the ETH address;

Reply:
Right, so no signers for Ethereum oracles as discussed.